lib/plugins/aws/provider/awsProvider.js

```diff
@@ -237,6 +237,56 @@ class AwsProvider {
             },
           },
         },
+        resources: {
+          properties: {
+            // Not replicating the full JSON schema from https://s3.amazonaws.com/cfn-resource-specifications-us-east-1-prod/schemas/2.15.0/all-spec.json
+            // as that gets into the specifics for each resource type.
+            //
+            // The only required attribute is `Type`; `Properties` and other common attributes are optional.
+            // See also https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resources-section-structure.html
+            Resources: {
```

Comment:
Let's provide definitions for all possible CF template properties (listed here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-anatomy.html ) but ofc without going into deep details, it should serve more as sanity check

Reply:
To clarify: you're just looking for a top-level check on `AWSTemplateFormatVersion`, `Description`, `Metadata`, `Parameters`, `Mappings`, `Conditions`, `Transform`, and `Outputs` in addition to the existing schema for `Resources` here?

Reply:
Yes, exactly, as all of them are supported by Framework